docs/user-guide/dsl.md

```diff
@@ -67,6 +67,29 @@ This may or may not be what you intended.
 !!! note
     Mixing decorated and undecorated functions may certainly be useful. __What's important is that you are able to distinguish between the things that will happen at build time and the things that will happen at runtime__.
 
+## Default Values and Hardcoded Values
+
+When defining DAGs and tasks, one can make use of python's default values. For example, the input of the DAG shown below has a default value, 1. The task `sum` has a default value, 2, for its second parameter, `b`. And this makes the DAG return 3 when invoked, if no arguments are passed.
+
+```python
+--8<-- "docs/code_snippets/default_values/imperative.py"
+```
+
+One can also use hardcoded values for tasks and DAGs. Modifying a bit the previous example, `b` has now a hardcoded value, 3, which is used in place of its default value, 2. This makes now return 3 when invoked, if no arguments are passed.
+
+```python
+--8<-- "docs/code_snippets/hardcoded_values/imperative.py"
+```
+
+A more sophisticated example, a nested DAG with both default and hardcoded values, is shown below.
+
+```python
+--8<-- "docs/code_snippets/default_hardcoded_values/imperative.py"
+```
```

Comment:
How about we indicate what the output of running each example would be? Just to make it extra clear that the behavior is the one the users are expecting

Reply:
done, resolving